### DISCUSSION TITLE
Clarification on Copilot NeoVim Context

### DISCUSSION THREAD

**Discussion by forrestfwilliams** [community/community#89846] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hello, in presentations I've viewed on how to make the best use of GitHub Copilot, the team has mentioned that Copilot uses context from 1-2 other open files in your editor to inform its suggestions. However, this was always discussed in the context of VS Code, and I'm not sure if the same is true for the NeoVim Copilot plugin.

When using the NeoVim Copilot plugin, does it gain context from your other open buffers, or is its context constrained to the currently active buffer?

**Comment by thanhdatvo** [comment]
I got the same issue.
When I use copilot in VSCode, it get context of the project for suggestion.
In neovim, it looks like only use the context of the current active file for suggestion.

**Comment by tpope** [comment, marked as answer]
It uses context from other buffers. Make sure you're on the latest version.

**Reply by dherrera1911** [reply]
Excuse my vim ignorance. Will Copilot use files I have open in Vim across tmux splits?

**Reply by tpope** [reply]
No, it only knows about the current Vim instance.

**Reply by DaanJG** [reply]
Hi! Does it also draw context from terminal buffers? I use Neovim plugins that open a Python or R REPL in a terminal buffer and send bits of code from the buffer I'm editing code in to the terminal buffer. I would be very usefull if Copilot could read the output of my bits of code in the terminal buffer to suggest new code in the buffer I am editing.

Thanks!
